### PR TITLE
add lib key 'public.openTypeCategories' and remove from glyph lib

### PR DESCRIPTION
**Pull request by moyogo**
See [Support for storing glyph category #146](https://github.com/unified-font-object/ufo-spec/issues/146) and in particular the discussion under https://github.com/unified-font-object/ufo-spec/issues/146#issuecomment-768210154.

### CODE REVIEW

**Comment by justvanrossum** [comment]
Looks good to me, but now I wonder: should it perhaps be 'public.openTypeCategories', in parallel to 'public.postscriptNames' being plural?

**Review by justvanrossum — APPROVED** [review]

**Comment by benkiel** [comment]
@moyogo one question before merge: can a glyph have more than one category? We should be explicit if this is one-to-one or one-to-many.

Edit: I'm thinking of marks that are also bases, reading the OT spec now for clarity.

**Comment by benkiel** [comment]
One other thing to add: it should be stated that the glyph does not need to be in the font (like what we state for glyph order, i.e.: `Glyph names in the font may not appear in the order. The order may contain glyphs that are not in the font.`)

File: versions/ufo3/lib.plist.md
Comment (benkiel) [inline versions/ufo3/lib.plist.md:44]: Should state the at the glyph name may not be in the font, and that all glyphs in the font do not need to appear in the list.

**Comment by benkiel** [comment]
@moyogo The OT spec doesn't say if glyphs can be in more than one class (thanks OT spec). I defer to folks more well versed than me here, but we should state what is possible (only one class or multiple classes allowed).

**Comment by justvanrossum** [comment]
The GDEF Glyph Class Definition Table only allows one class per glyph (this is at least implied by the Class Definition Table Format 1 and 2 data structures). So I think it makes sense to only allow a single category per glyph, here, too.

**Comment by moyogo** [comment]
> @moyogo one question before merge: can a glyph have more than one category? We should be explicit if this is one-to-one or one-to-many.

Good point. The [Glyph Class Definition Table](https://docs.microsoft.com/en-us/typography/opentype/spec/gdef#glyph-class-definition-table) structure only allows a glyph to be in one category as it is either structured as a Class Definition Table Format 1 (an array of glyph class ordered by glyphID) or a Class Definition Table Format 2 (tuples of glyph ranges and their glyph class -- ranges cannot overlap).

> One other thing to add: it should be stated that the glyph does not need to be in the font (like what we state for glyph order, i.e.: `Glyph names in the font may not appear in the order. The order may contain glyphs that are not in the font.`)

Good point. I'll add that.

**Comment by benkiel** [comment]
Got it, sorry for not groking that distinction of one class table (not multiple).

File: versions/ufo3/lib.plist.md
Comment (benkiel) [reply versions/ufo3/lib.plist.md:44]: would say then "must be only one of" and "glyph names can only appear once" (yes, the data structure does imply this, I know)

**Review by benkiel — CHANGES_REQUESTED** [review]

**Review by benkiel — APPROVED** [review]

**Pull request merged**
